### ISSUE TITLE
Bad left/right values after building the tree and saving it

### ISSUE THREAD

**Issue by claustres** [groupdock/mongoose-nested-set#13]
I've just tested your plugin that works well until I try to save the elements of my tree after calling rebuildTree(). It seems that the lft/rgt field are updated in 'strange' way afterwards. For example I have a root node with the correct lft=1 and rgt=14 (my tree has 7 elements) and after saving all elements the values are lft=1 and rgt=26...

I noticed the pre middleware for saving in your plugin but don't exactly understand what it is supposed to do.

Any help is welcome.

Thanks !

**Comment by stavros-zavrakas** [comment]
Did you find a solution to that @claustres? It seems that I am experiencing the same problem. The left and the right values of the tree are fine but when I execute an update on a document using the mogoose save method the pre save hook destroys the left and the right values at least of that specific document that I am trying to update. Is it possible that the save method should not be used to update the document?

**Comment by claustres** [comment]
Yes but it was a really big hack, it seems that the pre save method updates rgt/lft value wrongly, don't know why but thus I removed it. In my model where I use the plugin I added the following by finding how Mongoose stores the methods on the schema:
```javascript
_.remove(SystemSchema.callQueue, function(item) { return (item[0] === 'pre') && (item[1]['0'] === 'save'); });
```
`remove` is a function from the lodash library that simply removes all element of an array according to a predicate (https://lodash.com/docs#remove), you can also do it with pure JS.

**Comment by stavros-zavrakas** [comment]
Thanks claustres.
I think that maybe the purpose of the save functionality is not to update a document. In my case that was the scenario that was breaking. So I tried to split the cases and if I need to update a document to update using the mongoose update function instead of the save.
